### PR TITLE
feat(tools): Python script for automatically applying version (#6012)

### LINKED ISSUE #6012: Add a release script
### Problem to solve

We need a reliable was to update all the version numbers in
- `lv_conf_template.h`
- `lvgl.h`
- `library.json`
- `library.properties`
-  `lvgl.pc.in`

### Success criteria

There is script which updates all these

### Solution outline

Having a Python script with some `sed` calls seems simple. It could work like:
```
python3 update_version.py 9.1.2
python3 update_version.py 9.1.2-dev
```

### Rabbit holes

Can't think of anything.

### Testing

Hard to test...

### Teaching

Internal tool, having comments at beginning of the file and a `--help` should be enough.

### Considerations

_No response_

### PR DESCRIPTION
**Pull request by becseya**
Fixes #6012

### CODE REVIEW

File: scripts/update_version.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import argparse
+from collections import namedtuple
+
+
+def get_arg():
+    parser = argparse.ArgumentParser(
+        description='Apply the specified version to affected source files')
+    parser.add_argument('version', metavar='version', type=str,
+                        help='The version to apply')
+
```
Comment (kisvegabor) [inline scripts/update_version.py:16]: Please add a `help` too with usage example(s).

File: scripts/update_version.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    return parser.parse_args()
+
+
+class Version:
+    RE_PATTERN = r"(\d+)\.(\d+)\.(\d+)(-dev)?"
```
Comment (kisvegabor) [inline scripts/update_version.py]: Do not limit it to `-dev`.
```suggestion
    RE_PATTERN = r"(\d+)\.(\d+)\.(\d+)(-a-zA-Z0-9)?"
```

**Review by kisvegabor — COMMENTED** [review]
Looks good, just a few minor comments. 

When https://github.com/lvgl/lvgl/pull/5865 will be merged this script needs to be slightly updated. So let's wait for https://github.com/lvgl/lvgl/pull/5865.

File: scripts/update_version.py (hunk 1 above)
Comment (becseya) [reply scripts/update_version.py:16]: added

File: scripts/update_version.py (hunk 2 above)
Comment (becseya) [reply scripts/update_version.py]: updated

**Comment by kisvegabor** [comment]
@becseya #5865 is merged, please update this PR accordingly.

**Comment by becseya** [comment]
Updated

**Review by kisvegabor — COMMENTED** [review]
I've tested it and it works very well! Thank you!

**Comment by becseya** [comment]
Does it need to be cherry-picked to other release branches?

**Comment by kisvegabor** [comment]
I think no for now. `v9.2` will be released from `master` and it's unlikely that we will have `v9.1.1`. There no plan for `v8.4.1` either.

File: scripts/update_version.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import argparse
+from collections import namedtuple
```
Comment (XuNeo) [inline scripts/update_version.py]: Is `namedtuple` not used?

**Review by XuNeo — DISMISSED** [review]
I'm not familiar with regular expression. Please give it a full test.

**Comment by kisvegabor** [comment]
How can we test such a script?

File: scripts/update_version.py (hunk 3 above)
Comment (becseya) [reply scripts/update_version.py]: Good catch, removed it

**Comment by becseya** [comment]
It was tested manually for several version variants. I'd go with that, writing test cases for such a small script feels extensive.

**Review by FASTSHIFT — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Comment by kisvegabor** [comment]
Not all questions are clear about updating version numbers but I merge it now to make some order. 

Probably a follow up PR will be required.

**Pull request merged**